### ISSUE TITLE
When commands share the same label, the latter gets executed

### ISSUE THREAD

**Issue by akicho8** [magit/transient#304]
This library is extremely useful, and I regularly rely on it, even outside of Magit.

However, I encountered an issue with the following code snippet. When I press `a` the action associated with `c2` is executed instead of `c1`.

```
(transient-define-prefix f1 ()
  [
   ("a" "item1" (lambda () (interactive) (message "c1")))
   ("b" "item1" (lambda () (interactive) (message "c2")))
   ])
```

## Expected Behavior

I expect that when I press `a` the action associated with c1 should be executed.

## Actual Behavior

Currently, pressing `a` causes the action associated with c2 to be executed instead.

## Additional Context

I'm aware of this pitfall and can usually avoid it, but I occasionally fall into this trap. It would be very helpful if this could be fixed so that pressing `a` correctly triggers `c1`.

**Comment by tarsius** [comment]
Thanks for the report!
Should be fixed with b54af9e4c40c5fea6781d0279b99ed788c9d8edd.

**Comment by akicho8** [comment]
Thank you for the prompt response!
The revised code worked flawlessly.
With this, transient.el has become a perfect library.
